### PR TITLE
Bug 1250030 - Stop throw in PromiseStorage if no indexedDB, r=timdream

### PR DESCRIPTION
**Pull request by raylin**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=379a595b8ac5d2560779940aa21215119fd33e45

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=6747ab67fed18de424d8511d15d3161c55bea2e0

File: apps/keyboard/test/unit/shared/promise_storage_test.js [hunk 1]
```diff
@@ -38,9 +38,9 @@ suite('PromiseStorage', function() {
       });
 
       p.then(function() {
-          assert.isTrue(false, 'should not resolve');
+          assert.isTrue(true, 'should passed even PromiseStorage start failed');
```
Comment (timdream) [inline apps/keyboard/test/unit/shared/promise_storage_test.js]: I don't think this task case should change.

File: apps/keyboard/js/shared/promise_storage.js [hunk 2]
```diff
@@ -40,7 +40,7 @@ PromiseStorage.prototype.start = function() {
   return this._getDatabase().catch(function(e) {
     this._openPromise = null;
 
-    throw e;
+    e && console.error(e);
```
Comment (timdream) [inline apps/keyboard/js/shared/promise_storage.js]: You would hide other errors here, as shown by the test case. The goal of this patch is only about stop throwing/rejecting for the non-exist Indexed DB case.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=b8bb138ad71d64a5428b35338641cc74d3493221

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=dd3ec9882284e9b52e10fd1ed2857c0a2a816a08

File: apps/keyboard/test/unit/shared/promise_storage_test.js [hunk 3]
```diff
@@ -17,17 +17,51 @@ suite('PromiseStorage', function() {
...
     name = 'storage';
+  });
 
-    mockIndexedDB = new MockIDBFactory();
-    this.sinon.spy(mockIndexedDB, 'open');
+  suite('unsupport indexedDB', function() {
+    setup(function() {
+      window.indexedDB.open = null;
+      storage = new PromiseStorage(name);
+      storage._supportIndexedDB = false;
```
Comment (timdream) [inline apps/keyboard/test/unit/shared/promise_storage_test.js]: Leave a `XXX` comment on top of this line.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=76e336c293458d09a1bca4cac01fe4905f42af6a

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=eb0d05cfae10981f1ab977feff7dcbe401186d60

**Pull request merged**